viewing_party/party.py

```diff
@@ -1,23 +1,188 @@
+    return user_updated_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+# Calculates average rating for watched movies
+def get_watched_avg_rating(user_data):
+    counter = 0.0
+    for key, value in user_data.items():
+        for information in value:
+            counter += information["rating"]
+
+    length_of_movies = len(user_data["watched"])
+    print(length_of_movies)
+    if length_of_movies == 0:
+        return counter
+    else:
+        average = counter / length_of_movies
+        return average
+
+# Finds most watche genre in user's watched within dictionary
+
+
+def get_most_watched_genre(user_data):
+    most_watched_genre = {}
+    for movies in user_data["watched"]:
+        for values in movies:
+            if movies["genre"] not in most_watched_genre:
+                most_watched_genre[movies["genre"]] = 1
+            else:
+                most_watched_genre[movies["genre"]] += 1
+    length_of_movies = len(user_data["watched"])
+    if length_of_movies == 0:
+        return None
+    else:
+        most_watched_genre = max(
+            most_watched_genre, key=most_watched_genre.get)
+        return most_watched_genre
+
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+
+def get_unique_watched(user_data):
+    user_unique_watched = []
+    user_watched = []
+    friends_watched = []
+
+    for movies in user_data["watched"]:
+        user_watched.append(movies)
+
+    for movies in user_data["friends"]:
+        for details in movies["watched"]:
+            friends_watched.append(details)
+
+    user_unique_watched = [i for i in user_watched if i not in friends_watched]
+
+    if len(user_unique_watched) == []:
+        return 0
+    else:
+        return user_unique_watched
+
+
+def get_friends_unique_watched(user_data):
+    friends_unique_watched = []
+    user_watched = []
+    friends_watched = []
+
+    for movies in user_data["watched"]:
+        user_watched.append(movies)
+
+    for movies in user_data["friends"]:
+        for details in movies["watched"]:
+            friends_watched.append(details)
+
+    friends_unique_watched = [
+        i for i in friends_watched if i not in user_watched]
+
+    no_duplicates_movie_list = []
+    for i in friends_unique_watched:
+        if i not in no_duplicates_movie_list:
+            no_duplicates_movie_list.append(i)
+
+    if len(friends_unique_watched) == []:
+        return 0
+    else:
+        return no_duplicates_movie_list
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    recommend_list = []
+    combined_list = []
+    subscriptions = []
+    for movies in user_data["subscriptions"]:
+        subscriptions.append(movies)

```

Comment:
I don't think we need this. What is inside `user_data["subscriptions"]`? If we check the `test_constants.py` file, we see it is not movies. It is a list of streaming services, so we are re-creating a list we already have into a new variable.

If we want the variable because it's easier to read, then perhaps instead we could do:

```suggestion
    subscriptions = user_data["subscriptions"]
```